Sofa/Component/IO/Mesh/tests/MeshSTLLoader_test.cpp

```diff
@@ -0,0 +1,93 @@
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+
+#include <sofa/testing/BaseTest.h>
+#include <sofa/helper/system/FileRepository.h>
+
+#include <SofaGeneralLoader/MeshSTLLoader.h>
+#include <sofa/helper/BackTrace.h>
+
+using namespace sofa::component::loader;
+using sofa::testing::BaseTest;
+using sofa::helper::BackTrace;
+
+namespace sofa
+{
+namespace meshstlloader_test
+{
+
+    int initTestEnvironment()
+    {
+        BackTrace::autodump();
+        return 0;
+    }
+    int s_autodump = initTestEnvironment();
+
+
+    class MeshSTLLoaderTest : public BaseTest,
+                              public MeshSTLLoader
+    {
+    public:
+
+        MeshSTLLoaderTest()
+        {}
+
+        /**
+         * Helper function to check mesh loading.
+         * Compare basic values from a mesh with given results.
+         */
+        void loadTest(std::string filename, int nbPositions, int nbEdges, int nbTriangles, int nbQuads, int nbPolygons,
+                      int nbTetra, int nbHexa, int nbNormals)
+        {
+            this->setFilename(sofa::helper::system::DataRepository.getFile(filename));
+
+            EXPECT_TRUE(this->load());
+            EXPECT_EQ((size_t)nbPositions, this->d_positions.getValue().size());
+            EXPECT_EQ((size_t)nbEdges, this->d_edges.getValue().size());
+            EXPECT_EQ((size_t)nbTriangles, this->d_triangles.getValue().size());
+            EXPECT_EQ((size_t)nbQuads, this->d_quads.getValue().size());
+            EXPECT_EQ((size_t)nbPolygons, this->d_polygons.getValue().size());
+            EXPECT_EQ((size_t)nbTetra, this->d_tetrahedra.getValue().size());
+            EXPECT_EQ((size_t)nbHexa, this->d_hexahedra.getValue().size());
+            EXPECT_EQ((size_t)nbNormals, this->d_normals.getValue().size());
+        }
+
+    };
+
+    /** MeshGmshLoader::load()
+    * For a given meshes check that imported data are correct
+    */
+    TEST_F(MeshSTLLoaderTest, LoadCall)
+    {
+        //Check loader high level result for one file
+        this->setFilename(sofa::helper::system::DataRepository.getFile("mesh/circle_knot_ascii.stl"));
+        EXPECT_TRUE(this->load());
```

Comment:
`EXPECT_TRUE(this->load()); << "Added a failing test because XXXXXX, to be fixed."`